aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -95,22 +98,38 @@ public Orchestrator(
      * @return A Completable operation to start the sync engine orchestrator
      */
     @NonNull
-    public Completable start() {
-        return Completable.fromAction(() -> {
-            storageObserver.startObservingStorageChanges();
-            subscriptionProcessor.startSubscriptions();
-            syncProcessor.hydrate().blockingAwait();
-            mutationProcessor.startDrainingMutationOutbox();
-            subscriptionProcessor.startDrainingMutationBuffer();
+    public synchronized Completable start() {
+        return Completable.fromSingle(single -> {
```

Comment:
Would this be better as just:
```
Completable.create(emitter -> {
    try {
        tryToSucceed();
        emitter.onComplete();
    } catch (WhateverException whatever) {
       emitter.onError(whatever);
    } 
});
```

`fromCallable` or `fromAction` or `forRunnable` ... one of these factories must do the exception handling for you, I think, too?

like,
```
Completable.fromAction(() -> {
}).doOnError(thrownInsideOfAction -> {
    // I need to look up if it works like this ...
});
```